server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -152,6 +169,174 @@ public DenseVectorFieldMapper build(MapperBuilderContext context) {
+
+            @Override
+            void checkVectorBounds(float[] vector) {
+                StringBuilder errorBuilder = null;
+
+                for (int index = 0; index < vector.length; ++index) {
+                    float value = vector[index];
+                    double floor = Math.floor(value);
+
+                    if (value - floor > 1e-4 || value < Byte.MIN_VALUE || value > Byte.MAX_VALUE) {
+                        errorBuilder = new StringBuilder(
+                            this
+                                + " element_type vectors only support integers between"
+                                + "["
+                                + Byte.MIN_VALUE
+                                + ", "
+                                + Byte.MAX_VALUE
+                                + "] but found ["
+                                + value
+                                + "] at index ["
+                                + index
+                                + "];"
+                        );
+                        break;
+                    }
+                }
+
+                if (errorBuilder != null) {
+                    throw new IllegalArgumentException(appendErrorElements(errorBuilder, vector).toString());
+                }
+            }
+
+            @Override
+            void checkVectorMagnitude(VectorSimilarity similarity, float[] vector, float squaredMagnitude) {
+                StringBuilder errorBuilder = null;
+
+                if (similarity == VectorSimilarity.cosine && Math.sqrt(squaredMagnitude) == 0.0f) {
+                    errorBuilder = new StringBuilder(
+                        "The [" + VectorSimilarity.cosine.name() + "] similarity does not support vectors with zero magnitude."
+                    );
+                }
+
+                if (errorBuilder != null) {
+                    throw new IllegalArgumentException(appendErrorElements(errorBuilder, vector).toString());
+                }
+            }
+        },
+
+        FLOAT32(4) {
+
+            @Override
+            public String toString() {
+                return "float";
+            }
+
+            @Override
+            KnnVectorField createKnnVectorField(String name, float[] vector, VectorSimilarityFunction function) {
+                return new KnnVectorField(name, vector, function);
+            }
+
+            @Override
+            IndexFieldData.Builder fielddataBuilder(DenseVectorFieldType denseVectorFieldType, FieldDataContext fieldDataContext) {
+                return new VectorIndexFieldData.Builder(
+                    denseVectorFieldType.name(),
+                    CoreValuesSourceType.KEYWORD,
+                    denseVectorFieldType.indexVersionCreated,
+                    denseVectorFieldType.dims,
+                    denseVectorFieldType.indexed
+                );
+            }
+
+            @Override
+            void checkVectorBounds(float[] vector) {
+                // nothing to check
```

Comment:
Maybe we should check for `isFinite` and `!isNaN`?

Reply:
Let me investigate what happens now under a NaN condition to ensure there are no bwc issues, but I do like this idea!

Reply:
So, it appears the jackson parser doesn't allow NaN values to be parsed currently (https://github.com/elastic/elasticsearch/issues/2863). I could add these checks and unit tests as part of float anyway since it sounds like we may want to enable the feature at some point.

Reply:
I added additional checks for `NaN` and infinite for both `float` and `byte` `element_type`s. For now these are typically caught during parsing, but if (#2863) is resolved these checks become relevant.